Python/Product/Debugger/PtvsdVersionHelper.cs

```diff
@@ -51,18 +51,23 @@ internal class PtvsdVersionHelper {
             if (PackageVersion.TryParse(response.Debugger.Version, out PackageVersion runningVersion)) {
                 var bundledPtvsdVersion = PackageVersion.Parse(PtvsdVersion.Version);
                 if (runningVersion.CompareTo(bundledPtvsdVersion) < 0) {
-                    ThreadHelper.JoinableTaskFactory.RunAsync(async () => {
-                        await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-                        MessageBox.Show(
-                            new Win32Window(Process.GetCurrentProcess().MainWindowHandle),
-                            Strings.InstalledPtvsdOutdatedMessage.FormatUI(response.Debugger.Version, PtvsdVersion.Version),
-                            Strings.ProductTitle,
-                            MessageBoxButtons.OK,
-                            MessageBoxIcon.Warning);
-                    });
+                    ShowPtvsdVersionMessage(Strings.InstalledPtvsdOutdatedMessage.FormatUI(response.Debugger.Version, PtvsdVersion.Version));
                 }
             }
         }
+
+        public static void VerifyPtvsdVersionError(PtvsdVersionArguments args, ProtocolException ex) {
+            ShowPtvsdVersionMessage(Strings.InstalledPtvsdOutdatedMessage.FormatUI("unknown", PtvsdVersion.Version));
+        }
+
+        private static void ShowPtvsdVersionMessage(string message) {
+            MessageBox.Show(
```

Comment:
You no longer need to switch to main thread to avoid UI freezing?

Reply:
Nope, message box does not require that. Explicitly switching to UI thread was blocking the message box from getting displayed in cases where there was pending operation going on while we were attempting to show the message box.